<launch>
  <node name="nodo_sensor_lidar"
        pkg="ydlidar_ros_driver"
        type="ydlidar_ros_driver_node"
        output="screen"
        respawn="false">
    <param name="port"           type="string" value="/dev/ydlidar"/>
    <param name="frame_id"       type="string" value="laser_frame"/>
    <param name="ignore_array"   type="string" value=""/>
    <param name="baudrate"       type="int"    value="128000"/>
    <param name="lidar_type"     type="int"    value="1"/>
    <param name="device_type"    type="int"    value="0"/>
    <param name="sample_rate"    type="int"    value="5"/>
    <param name="abnormal_check_count" type="int"    value="4"/>
    <param name="resolution_fixed" type="bool"   value="true"/>
    <param name="auto_reconnect" type="bool"   value="true"/>
    <param name="reversion"      type="bool"   value="false"/>
    <param name="inverted"       type="bool"   value="true"/>
    <param name="isSingleChannel" type="bool"   value="false"/>
    <param name="intensity"      type="bool"   value="false"/>
    <param name="support_motor_dtr" type="bool"   value="true"/>
    <param name="invalid_range_is_inf" type="bool"   value="false"/>
    <param name="point_cloud_preservative" type="bool"   value="false"/>
    <param name="angle_min"      type="double" value="-180" />
    <param name="angle_max"      type="double" value="180" />
    <param name="range_min"      type="double" value="0.1" />
    <param name="range_max"      type="double" value="12.0" />
    <param name="frequency"      type="double" value="10.0"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser"
        args="0.0 0.0 0.2 0.0 0.0 0.0 /base_footprint /laser_frame 20" />

  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <param name="scan_topic" value="/scan"/>
    <param name="base_frame" value="/base_footprint"/>
    <param name="odom_frame" value="/odom"/>
    <param name="map_frame" value="/map"/>
    <rosparam command="load" file="$(find base_robot)/config/gmapping_params.yaml"/>
  </node>

  <node pkg="map_server" type="map_saver" name="map_saver" output="screen" args="-f $(find base_robot)/maps/my_map" />

  <rosparam command="load" ns="local_costmap" file="$(find base_robot)/config/costmap_params.yaml"/>
  <node pkg="costmap_2d" type="costmap_2d_node" name="local_costmap" ns="local_costmap">
    <rosparam file="$(find base_robot)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
  </node>

   <node pkg="rosserial_python" type="serial_node.py" name="Arduino_node" output="screen">
    <param name="port" value="/dev/ttyACM0"/> <param name="baud" value="57600"/>
  </node>

</launch>